src/datadog/trace_sampler_config.cpp

```diff
@@ -196,6 +197,9 @@ Expected<FinalizedTraceSamplerConfig> finalize_config(
     result.metadata[ConfigName::TRACE_SAMPLING_RATE] = ConfigMetadata(
         ConfigName::TRACE_SAMPLING_RATE, to_string(*sample_rate, 1),
         ConfigMetadata::Origin::CODE);
+  } else {
+    result.metadata[ConfigName::TRACE_SAMPLING_RATE] = ConfigMetadata(
+        ConfigName::TRACE_SAMPLING_RATE, "1.0", ConfigMetadata::Origin::CODE);
```

Comment:
Wanted `ConfigMetadata::Origin::DEFAULT` here? Not sure if it has any side effect, and it's true that it is `CODE` as well though

Reply:
Good catch. I did it right but I must have lost that part during a rebase :/

Reply:
```suggestion
        ConfigName::TRACE_SAMPLING_RATE, "1.0", ConfigMetadata::Origin::DEFAULT);
```